devops/jobs/ImageBuilder.groovy

```diff
@@ -26,54 +26,61 @@ import static org.edx.jenkins.dsl.Constants.common_wrappers
 
 class ImageBuilder {
     public static def job = { dslFactory, extraVars ->
-        dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/image-builder") {
+        //assertions come earlier to ensure APPS_TO_CONFIG can be read
+        assert extraVars.containsKey('APPS_TO_CONFIG'): "Please define APPS_TO_CONFIG"
+        assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
 
-            assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+        // for each application
+        extraVars.get("APPS_TO_CONFIG").each { app_name, app_config ->
+            dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/image-builders/" + app_name + "-image-builder") {
```

Comment:
This will require an update to the internal documentation to include creation of this folder: edx/jenkins-job-dsl-internal#126.